cmd/initializer/main.go

```diff
@@ -15,141 +15,32 @@
 	goflag "flag"
-	"fmt"
 	"os"
-	"strconv"
-	"strings"
 
-	"github.com/BurntSushi/toml"
 	"github.com/spf13/pflag"
 	"k8s.io/klog/v2"
-)
-
-type options struct {
-	configPath          string
-	defaultConfigPath   string
-	datanodeRPCPort     int32
-	datanodeServiceName string
-	namespace           string
 
-	// The storage options.
-	storageType     string
-	localStorageDir string
-	bucket          string
-	prefix          string
-	accessKeyID     string
-	secretAccessKey string
-
-	endpoint string
-	region   string
-}
+	"github.com/GreptimeTeam/greptimedb-operator/cmd/initializer/internal"
+)
 
 func main() {
-	var opts options
-
-	pflag.StringVar(&opts.configPath, "config-path", "/etc/datanode/datanode.toml", "default config path")
-	pflag.StringVar(&opts.defaultConfigPath, "input-config-file", "/datanode/defaults/datanode.toml", "output config path")
-	pflag.StringVar(&opts.datanodeServiceName, "datanode-service-name", "", "the name of datanode service")
-	pflag.StringVar(&opts.namespace, "namespace", "", "the cluster namespace")
+	var opts internal.Options
 
-	// Handle storage options.
-	pflag.StringVar(&opts.storageType, "storage-type", "", "the storage type")
-	pflag.StringVar(&opts.localStorageDir, "local-storage-dir", "", "the directory of local storage")
-	pflag.StringVar(&opts.bucket, "bucket", "", "the bucket of s3 storage")
-	pflag.StringVar(&opts.prefix, "prefix", "", "the prefix of s3 storage")
-	pflag.StringVar(&opts.accessKeyID, "access-key-id", "", "the access key id of s3 storage")
-	pflag.StringVar(&opts.secretAccessKey, "secret-access-key", "", "the secret access key of s3 storage")
-	pflag.StringVar(&opts.endpoint, "endpoint", "", "the endpoint of s3 storage")
-	pflag.StringVar(&opts.region, "region", "", "the region of s3 storage")
+	pflag.StringVar(&opts.ConfigPath, "config-path", "/etc/greptimedb/config.toml", "the config path")
+	pflag.StringVar(&opts.InitConfigPath, "init-config-path", "/etc/greptimedb/init-config.toml", "the init config path")
+	pflag.StringVar(&opts.Namespace, "namespace", "", "the namespace of greptimedb cluster")
+	pflag.StringVar(&opts.ComponentKind, "component-kind", "", "the component kind")
 
-	// FIXME(zyy17): The greptimedb should support inject configs as env.
-	pflag.Int32Var(&opts.datanodeRPCPort, "datanode-rpc-port", 4001, "datanode rpc port")
+	pflag.StringVar(&opts.DatanodeServiceName, "datanode-service-name", "", "the name of datanode service")
+	pflag.Int32Var(&opts.DatanodeRPCPort, "datanode-rpc-port", 4001, "the datanode RPC port")
 
 	klog.InitFlags(nil)
 	pflag.CommandLine.AddGoFlagSet(goflag.CommandLine)
 
 	pflag.Parse()
 
-	// TODO(zyy17): We should add the flag to distinguish different components, not just datanode.
-	var datanodeConfig DatanodeConfig
-	_, err := toml.DecodeFile(opts.defaultConfigPath, &datanodeConfig)
-	if err != nil {
-		klog.Fatalf("Parse input toml file failed, err '%v'", err)
-	}
-
-	if opts.storageType == "S3" {
-		datanodeConfig.StorageConfig.Type = "S3"
```

Comment:
question: Users use ak/sk to access aws s3, there is no interface, only to modify the toml file to achieve.

Reply:
User still can use secret to setup their S3 credentials.